osu.Game/Screens/OnlinePlay/Multiplayer/Match/MultiplayerSpectateButton.cs

```diff
@@ -121,11 +114,7 @@ private void updateState()
 
         private void checkForAutomaticDownload()
         {
-            PlaylistItem? item = SelectedItem.Value;
-
-            downloadCheckCancellation?.Cancel();
```

Comment:
https://github.com/ppy/osu/pull/32250/files#r1998396760

> Why was this moved further down after all of the early returns?

The order of operations here makes no sense in the context of multiplayer and updating w.r.t `RoomUpdated`. In-fact, I would be surprised if this is not already causing problems - for example should we cancel the download check when a user's beatmap download progress is changed?

Reply:
To keep it simple, I've reverted the changes here in https://github.com/ppy/osu/pull/32460/commits/6f1c00ff97c83e150491efe2b773e2a0616091e5.

I still believe we're going to have to revisit this because I struggle to imagine this isn't already causing problems as I mentioned above. I mostly did it because it looked plain wrong but that's for a separate PR when required, if ever. It certainly isn't required for this batch of changes.